### DISCUSSION TITLE
Working example of multiple dev containers for a monorepo  ?

### DISCUSSION THREAD

**Discussion by KingBain** [devcontainers/community#12] [Q&A]
Looking to see if someone has a working example of multiple devcontainers for a monorepo. I have a repo with multiple projects and would like to spin up different containers depending on the project. 

I've read the documentation here: 

- https://docs.github.com/en/enterprise-cloud@latest/codespaces/setting-up-your-project-for-codespaces/introduction-to-dev-containers#devcontainerjson
- https://github.blog/2022-04-20-codespaces-multi-repository-monorepo-scenarios/
- https://www.youtube.com/watch?v=o5coAL7oE0o&list=PLj6YeMhvp2S6GjVyDHTPp8tLOR0xLGLYb&index=5

but I haven't been able to make whats documented work and I haven't found an example repo to expand on.

Does anyone have an example of it working available to share or a tutorial I can follow ?

TIA

**Comment by Chuxel** [comment]
This should work in Codespaces already.  Are you trying to use this in the Dev Containers extension?  If so, https://github.com/microsoft/vscode-remote-release/issues/7548 is tracking this and @chrmarti has been looking into adding it.

**Comment by kvten** [comment]
## Final result:

![Screenshot](https://user-images.githubusercontent.com/103538771/215295619-7a59f9f3-d222-4aa6-a6ff-d054ffb82018.png)

---

## Requirements

- VS Code
- Docker (must be running)
- VS Code extensions:
   - Docker
   - Dev Containers
   - Docker Explorer (optional)

## Steps

1. create the dev container folders (*_Dev folders in picture above)
2. create a `Dockerfile` in each folder (must have this name)
3. modify contents of each `Dockerfile`

For each dev container folder:

1. click >< button in lower left corner of VS Code `Open a remote window`
2. click `Open folder in container`
3. click `use Dockerfile` and let it finish
4. install VS Code extensions (optional)

## To Use

All files in dev container folder will be shared between host and container. Open dev container folder in VS Code and start working.

## Comments

- I automated the folder & Dockerfile creation with the  `create.proj_multi_file` function from a Python package I wrote [link](https://github.com/kenf1/High-Yield)
- `.devcontainer` is created automatically

Hope this helps!

**Reply by chrmarti** [reply]
Note that https://github.com/microsoft/vscode-remote-release/issues/7548 is becoming available with the upcoming release of the Dev Containers extension. Codespaces already supports it.
